zilliqa/src/exec.rs

```diff
@@ -1368,11 +1418,18 @@ fn scilla_create(
         ));
     };
 
+    let init_data = ContractInit::new(init_data)?;
+
+    // We need to cache external libraries used in the current contract. Scilla checker needs to import them to check the contract.
```

Comment:
If I understand correctly - All the executed external libraries will live persistently in `ext_libs_cache_dir`?

Do we need to do this? What if instead we didn't cache anything and just populated the directory with the specific libraries that are required when we are running the transaction? 

I'd like to avoid adding caches if we can avoid it, since it is extra complexity and room for more bugs to appear (e.g. outdated contracts in the cache, name collisions, etc.)

Reply:
On the other hand, doing so might lead to significant slowdowns in execution speed (admittedly, not as significant as executing Scilla itself!)

Reply:
I'd prefer we implement the simple version first and add a cache if we prove that loading a contract's libraries is a bottleneck. My hunch is that it won't be, since we already need to spend a bunch of time writing other files (such as the contract we're actually executing) at transaction execution time anyway.

Reply:
@JamesHinshelwood The name might be misleading. My intention is not to cache a library contract for future transactions, except for the current one. I don’t verify the existence of the external library file before writing it to the cache directory, ensuring the code is always up-to-date. To address this, I can remove the external library contract files after each transaction execution.

@rrw-zilliqa To be stateless, we need to touch the Scilla HTTP server too, don’t we?

Reply:
@saeed-zil  - yes, sadly.

If it were me, I'd zap the whole directory after every contract execution - in fact, I'd probably go one further and create a new directory for each contract execution, and then clean them up after myself - this allows you to have a debugging "keep" flag so that you can see what library was executed with each contract (which is important because you might have a single txn with several contract executions each of which uses a different library with the same name).

It also means that your cleanup code is `tokio::fs::remove_dir()` , which is safer than writing it yourself (and saves you worrying about dotfiles, illegal filenames, half-written files which were interrupted by a node crash, etc. etc.)